### DISCUSSION TITLE
Bug: LORA's don't get unloaded properly.

### DISCUSSION THREAD

**Discussion by pflky** [vladmandic/sdnext#1190] [General]
I noticed that when moving from one LORA to the next, the previous LORA is still being applied. The only way to actually reset it is to load a different checkpoint. This might be a problem with the LORA or LYCORIS extension itself, or maybe some side effect of the token merging feature, or maybe some side effect of the "generate forever" feature. I did turn off the generate forever and did another generation, and the issue still persisted, but perhaps it's initially triggered by using the generate forever feature.  

I haven't extensively tested it, I just know for sure that for some reason LORA's do stay active even when no longer referenced in the prompt. I also don't have any settings active like caching multiple models, so something else funky is going on with how extra networks are being handled. I do believe I did try to pair a LORA with a LYCORIS, so maybe some conflict there could be causing odd behavior. I don't know, I'm just trying to think of possibilities that might narrow it down if it happens under a specific condition that might be hard to pinpoint, but it may also just happen by default and people just haven't noticed.

**Comment by vladmandic** [comment]
like you said, most likely it starts when you apply both lora and lyco as each module tries to restore model weights on unload and its overwritting each other. its a hard conceptual problem, i hope to solve it once i switch to unified lora/locon/lyco handling.
